susemanager-utils/susemanager-sls/src/beacons/pkgset.py

```diff
@@ -49,20 +42,19 @@ def beacon(config):
           pkgset:
             interval: 5
 
-    '''
-
-    global COOKIE_PATH
+    """
 
     ret = []
-    if os.path.exists(COOKIE_PATH):
-        with open(COOKIE_PATH) as ck_file:
+    for cookie_path in PKGSET_COOKIES:
+        if not os.path.exists(cookie_path):
+            continue
+        with open(cookie_path) as ck_file:
             ck_data = ck_file.read().strip()
             if __virtualname__ not in __context__:
                 __context__[__virtualname__] = ck_data
             if __context__[__virtualname__] != ck_data:
-                ret.append({
-                    'tag': 'changed'
-                })
+                ret.append({"tag": "changed"})
                 __context__[__virtualname__] = ck_data
+                break
```

Comment:
Is it correct to `break` only in one of the two branches?

Reply:
This break is preventing multiple events sending in case if salt-minion and venv-salt-minion are installed in the same time. Cookie files are in different locations for this case.

Reply:
No, it's better to prevent breaking in the condition above. As the system could have two cookie files on the system at the same time, but one for some reason was not updated, so it's safer to check the rest and fire the event if at least one was changed.

Reply:
I still don't get the difference between `__virtualname__ not in __context__` and `__context__[__virtualname__] != ck_data` when it comes to "stop on this cookie file or go through the rest".

If a system has 2 of these cookie files and why do we want to insert into both, but only update the first?

Reply:
So this part:
```
            if __virtualname__ not in __context__:
                __context__[__virtualname__] = ck_data
```
is storing the cookie data in the context, but this time we can read the other data from the other cookie file which could be different from the date we stored in this section and we will trigger the event with:
```
            if __context__[__virtualname__] != ck_data:
                ret.append({"tag": "changed"})
                __context__[__virtualname__] = ck_data
                break
```
but in the other iteration, so we actually using the cookie data from both possible files, but firing only one single event. The described case is not really common, mostly one cookie file is present on the system, but in case of using Salt Bundle with the original salt minion it's better to prevent such situation, if not we can lost at least one package change from the system.